### PR TITLE
A bunch of improvements

### PR DESCRIPTION
**Pull request by sergv**
I collected together few commits that accumulated over time. Notable improvements are:
- Fix build for ghc 7.8 and reflect that in cabal file
- Allow functions and function arguments to be named `pattern`
- Allow functions to be named `forall` (which is weird but ghc is OK with that)
- Collect tags from GADT records
- Add more tests
- Minor stylistic and other cleanups

### CODE REVIEW

File: src/FastTags/LexerTypes.hs [hunk 1]
```diff
@@ -168,15 +175,11 @@ fixChar c
...
-          ConnectorPunctuation  -> Just symbol
-          DashPunctuation       -> Just symbol
-          OtherPunctuation      -> Just symbol
-          MathSymbol            -> Just symbol
-          CurrencySymbol        -> Just symbol
-          ModifierSymbol        -> Just symbol
-          OtherSymbol           -> Just symbol
           Space                 -> Just space
-          _other                -> Nothing
+          other                 ->
```
Comment (elaforge) [inline src/FastTags/LexerTypes.hs]: Just FYI guard works in case branches too, e.g. other | Util.isSymbol ... -> ...

**Comment by elaforge** [comment]
Good stuff!

I have just one optional comment.  Actually I'm not sure how the workflow is supposed to work in github, because where I work the submitter is the one that pushes the change, the reviewer just says "ok go ahead."

Let me see if I can just give push access to you, then you can merge when you please.

File: src/FastTags/LexerTypes.hs (hunk 1 above)
Comment (sergv) [inline src/FastTags/LexerTypes.hs]: Yes, thanks for noting, but I thought that guard would be too wide there.

**Comment by sergv** [comment]
Thanks for push access, I'll do the merge. If anything else comes up I'll open a PR first.

**Pull request merged**